ragna/assistants/_ai21labs.py

```diff
@@ -0,0 +1,80 @@
+from typing import AsyncIterator, cast
+
+from ragna.core import RagnaException, Source
+
+from ._api import ApiAssistant
+
+
+class Ai21LabsAssistant(ApiAssistant):
+    _API_KEY_ENV_VAR = "AI21_API_KEY"
+    _MODEL_TYPE: str
+    _CONTEXT_SIZE: int = 8_192
+
+    @classmethod
+    def display_name(cls) -> str:
+        return f"AI21Labs/jurassic-2-{cls._MODEL_TYPE}"
+
+    @property
+    def max_input_size(self) -> int:
+        return self._CONTEXT_SIZE
+
+    def _make_system_content(self, sources: list[Source]) -> str:
+        instruction = (
+            "You are a helpful assistant that answers user questions given the context below. "
+            "If you don't know the answer, just say so. Don't try to make up an answer. "
+            "Only use the sources below to generate the answer."
+        )
+        return instruction + "\n\n".join(source.content for source in sources)
+
+    async def _call_api(
+        self, prompt: str, sources: list[Source], *, max_new_tokens: int
+    ) -> AsyncIterator[str]:
+        response = await self._client.post(
```

Comment:
Could you include a link to the API docs here that describe how to structure the request and parse the response?

Reply:
Done